### PR TITLE
run BPF tests in CI

### PR DESCRIPTION
**Pull request by NimrodAvni78**
## Summary

Run the `bpf/tests` C unit tests in CI.

The `bpf/tests` directory holds native C unit tests for BPF-side logic (`bpf_ssl_read_guard`, `bpf_http2_preface_bounds`, `bpf_kafka_large_buffer`, `bpf_postgres_detection`, `bpf_float64`, `test_trace_util`, `test_large_buffer_loop`, `test_failed_connect_event`), compiled natively against the `bpf/tests/bpfcore` stubs. Nothing ran them: no workflow referenced the directory, and its `Makefile` only compiled the binaries (the `all` / `%: %.c` rules never execute them). As a result these tests provided no automated protection.

### Change

- Add a `test` target to `bpf/tests/Makefile` that compiles **and runs** every test binary, failing on the first non-zero exit.
- Add a dedicated `.github/workflows/bpf_unit_tests.yml` workflow (ubuntu, install clang, `make -C bpf/tests test`), mirroring the `bpf_verifier.yml` pattern and scoped via `paths:` to `bpf/**` (and the workflow file), so it only runs when BPF sources change rather than on every code PR.

Newly added C tests (e.g. those introduced alongside BPF fixes) are then covered automatically.

## Validation

- [x] I have read and followed the [contributing guidelines](https://github.com/open-telemetry/opentelemetry-ebpf-instrumentation/blob/main/CONTRIBUTING.md)
- [ ] If this enhances / fixes / changes a core feature, I have updated the [features documentation](https://github.com/open-telemetry/opentelemetry-ebpf-instrumentation/blob/main/devdocs/features.md) and [support matrix](https://github.com/open-telemetry/opentelemetry-ebpf-instrumentation/blob/main/SUPPORT_MATRIX.md) as needed.

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -118,6 +118,24 @@ jobs:
             exit 1
           fi
 
+  bpf-unit-tests:
```
Comment (pinoOgni) [inline .github/workflows/pull_request.yml]: I know that these C tests are cheap but they will run on every code PR, not only for bpf changes because the paths filter is broad. 

A possible idea is to create `.github/workflows/bpf_unit_tests.yml` file mirroring the `.github/workflows/bpf_verifier.yml` pattern.

**Review by mariomac — APPROVED** [review]
Looks OK to me

**Review by mmat11 — APPROVED** [review]

File: .github/workflows/bpf_unit_tests.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+on:
+  push:
+    branches: [main, release-*]
+  pull_request:
+    branches: [main, release-*]
+    paths:
+      - ".github/workflows/bpf_unit_tests.yml"
+      - "bpf/**"
+  workflow_call:
```
Comment (pinoOgni) [inline .github/workflows/bpf_unit_tests.yml:11]: I think we should wire this block into `release.yml` like `bpf_verifier.yml` is and also add it to the `tests-passed` job's `needs:` list so that's blocks the release (like the `bpf-verifier:`).

**Review by pinoOgni — APPROVED** [review]
lgtm, I left a comment.
